### PR TITLE
Cleanup Windows Widgets Warnings

### PR DESCRIPTION
**Pull request by RobertBColton**
Here we go cleaning up the Win32 widget system's warnings. First, Josh's `gtklike.h` widgets were using misleading indentation (two if conditions on the same line with no braces). Second, `WINVER` was being redefined since the change where we added `get_window_handle()` so I just moved the definition to the top of the file before all `windows.h` includes.

### CODE REVIEW

File: ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp [hunk 1]
```diff
@@ -25,13 +25,14 @@
 **                                                                              **
 \********************************************************************************/
 
+// As is typical of Win32 code, this code is fuck-ugly. Refer to the GTK version for
+// porting to competent widget systems. Use this only for low-level APIs.
+#define WINVER 9001
```
Comment (fundies) [inline ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: Why did you move this define up? Now it will affect all the headers included below it

Comment (fundies) [reply ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: Also why are we even targeting a specific winver? shouldnt this be a #ifndef winver #define ....

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: This is a Win32 oddity fundies, it's supposed to be at the top before you include `windows.h`. That's what caused the warning, because if you don't define `WINVER`, then `windows.h` automatically defines it for you. All it does is "unlock" features of the open file dialog if I remember correctly.
https://docs.microsoft.com/en-us/windows/desktop/winprog/using-the-windows-headers

>The Windows header files use macros to indicate which versions of Windows support many programming elements. Therefore, you must define these macros to use new functionality introduced in each major operating system release.

Comment (fundies) [reply ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: well the you should move it before windows.h. not before our headers

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: No, because it's defined on a per-file basis and if any of our headers include `windows.h` (which was the issue... the `WINDOWShandle.h` I added includes `windows.h`). So, you are generally supposed to define it at the very top before all includes. This is the general rule of thumb and also what RGM is doing.

Comment (fundies) [reply ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: you're infecting files that don't use that macro directly. What if someone wants to use a different WINVER for whatever reason? You're just spreading the macro to all corners of our source like what happened with get_red/blue/green and our debug macros. If you want the macro defined globally at least only define it only one place. (probably the win32 makefile)

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: They shouldn't define it in a header - ever.

Comment (fundies) [reply ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: You can define it in one header and include that header. You shouldn't be defining it everywhere otherwise its a pain when it needs to be changed. Right now its two places but it could be 50 later

Comment (RobertBColton) [reply ENIGMAsystem/SHELL/Widget_Systems/Win32/widgets.cpp]: Thanks fundies, that would be more true if you were writing a header that's like helper methods to Win32. The BSD define of `WINVER` I think actually defines a different Windows version because IsmAvatar uses a different part of the Win32 API. Anyway, I'll talk to you later, my camera id card arrived so I'm going to get my new license
